lib/rules/sort-prop-types.js

```diff
@@ -261,6 +285,29 @@ module.exports = {
           }
         });
       },
+
+      TSTypeLiteral(node) {
+        if (node && node.parent.id) {
+          const currentNode = [].concat(
+            typeAnnotations.get(node.parent.id.name) || [],
+            node
+          );
+          typeAnnotations.set(node.parent.id.name, currentNode);
+        }
+      },
+
+      TSTypeAliasDeclaration(node) {
+        if (node.typeAnnotation.type === 'TSTypeLiteral' || node.typeAnnotation.type === 'ObjectTypeAnnotation') {
+          const currentNode = [].concat(
+            typeAnnotations.get(node.id.name) || [],
+            node.typeAnnotation
+          );
+          typeAnnotations.set(node.id.name, currentNode);
+        }
+      },
+
+      FunctionDeclaration: handleFunctionComponent,
+      ArrowFunctionExpression: handleFunctionComponent,
```

Comment:
maybe:
```suggestion
      FunctionDeclaration: checkTypes && handleFunctionComponent,
      ArrowFunctionExpression: checkTypes && handleFunctionComponent,
```

Reply:
Won't this give an error if `checkTypes` is falsy?

Reply:
Both (`checkTypes && handleFunctionComponent` and `checkTypes ? handleFunctionComponent : undefined`) of the above gives error, is following fine?
```js
FunctionDeclaration: checkTypes ? handleFunctionComponent : () => {},
ArrowFunctionExpression: checkTypes ? handleFunctionComponent : () => {},
```

Reply:
Doing the `Object.assign` way

Reply:
@ljharb this was done!